packages/terra-notification-dialog/src/_transientUtils.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { createContext } from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import Base from 'terra-base';
+
+const TransientComponentContext = createContext();
+
+/**
+ * Mounts a transient component to the document. The component will be wrapped by a Base component to provide i18n.
+ * @param {Component} component The component instance to be mounted.
+ * @param {String} containerId An id that will be used to create the element to which the component will be mounted.
+ * @param {String} locale The string locale used to load translated messages.
+ * @param {Object} customMessages The object containing custom messages for the given locale.
+ * @returns {Promise} A Promise is returned. If a component is already mounted for the given id value, the Promise will
+ *                    be rejected. Otherwise, the Promise will be resolved after the component has been mounted.
+ */
+const mount = (component, containerId, locale, customMessages) => {
+  /**
+   * If an element with the chosen id is already present, the new presentation is aborted.
+   */
+  if (document.getElementById(containerId)) {
+    return Promise.reject();
+  }
+
+  const mountElement = document.createElement('div');
+  mountElement.id = containerId;
+  mountElement.setAttribute('data-terra-transient-component', true);
+  document.body.appendChild(mountElement);
+
+  return new Promise((resolve) => {
+    render((
+      <Base locale={locale} customMessages={customMessages}>
+        <TransientComponentContext.Provider
```

Comment:
The provider is currently only exposing the containerId to child components so that those child components could unmount themselves, if necessary. 

I had toyed with the notion of providing a custom 'remove' function instead that would capture the id in an enclosure, but I wasn't sure if that was overcomplicating things.

Of course, we could clone the object direction on the component as a prop. It'd be straightforward for the ConfirmNotificationDialog's implementation, but anything more complex may benefit from the context implementation.